tag-register all raw arrays

guruhegde opened the pull request:
Made changes as suggested in #4101 

All tests pass.

vigsterkr commented:
@guruhegde thanks for the patch. i skid through the patch looks good... i've restarted the failed travis jobs, lets see the output and based on that we can merge it 👍

vigsterkr commented:
@guruhegde as you can see there some formatting issues in the patch:
https://travis-ci.org/shogun-toolbox/shogun/jobs/332627900#L740

you can use `clang-format` locally and check all the errors before sending in a fix.
see https://github.com/shogun-toolbox/shogun/wiki/Code-style for more details

guruhegde commented:
@vigsterkr sorry about the error. I will fix.

karlnapf commented on src/shogun/labels/MultilabelLabels.cpp:
```diff
@@ -76,13 +76,15 @@ CMultilabelLabels::init(int32_t num_labels, int32_t num_classes)
+	    this->m_parameters->add(&m_num_classes, "m_num_classes",
+	                            "Number of classes.");
+
+	    this->m_parameters->add_vector(&m_labels, &m_num_labels, "labels_array",
```
I think you don't need to change anything here.

guruhegde replied:
I will undo the changes.

karlnapf commented on src/shogun/modelselection/ParameterCombination.cpp:
```diff
@@ -43,8 +43,12 @@ CParameterCombination::CParameterCombination(CSGObject* obj)
+
 				p->add_vector((float64_t**)param->m_parameter, type.m_length_y,
 						param->m_name);
+				watch_param(
```
remove this for now

karlnapf replied:
(the class will be rewritten soon)

guruhegde replied:
Ok

karlnapf commented on src/shogun/transfer/multitask/MultitaskKernelPlifNormalizer.h:
```diff
@@ -328,10 +328,22 @@ class CMultitaskKernelPlifNormalizer: public CMultitaskKernelMklNormalizer
+
 		m_parameters->add_vector((SGString<float64_t>**)&distance_matrix, &num_tasksqr, "distance_matrix", "distance between tasks");
+		watch_param(
+			"distance_matrix", (SGString<float64_t>**)&distance_matrix,
```
I don't think this can work, these static casts don't make any sense
Could you remove you added watch_param statements here?

Then, in another PR, could you even comment out the add_vector calls?

guruhegde replied:
Ok.

karlnapf commented:
Pls rebase against develop

karlnapf commented on src/shogun/base/class_list.cpp.templ:
```diff
@@ -75,7 +75,7 @@ void shogun::delete_object(CSGObject* object)
 }
 
-std::set<std::string> available_objects()
+std::set<std::string> shogun::available_objects()
```
there are changes from another PR, you need to rebase.
Always check the diff !

karlnapf commented:
Sorry, something else was merged that partially addressed these changes. You will need to rebase and resolve. Otherwise this is good to merge soon

guruhegde commented on src/shogun/labels/MultilabelLabels.cpp:
```diff
@@ -76,13 +76,15 @@ CMultilabelLabels::init(int32_t num_labels, int32_t num_classes)
+	    this->m_parameters->add(&m_num_classes, "m_num_classes",
+	                            "Number of classes.");
+
+	    this->m_parameters->add_vector(&m_labels, &m_num_labels, "labels_array",
```
I removed tag register here, by mistake i didn't remove new line. hope it's ok.

guruhegde commented:
Addressed issues in the last commit. Tests pass. I hope this goes well.
Travis job may complain about style check in one of the files(as mentioned in IRC).

karlnapf commented:
tada! :)
Nice one, thanks so much for the patch

guruhegde commented:
Happy to see this! Thanks for the guidance :)

Questions:
Would you like me to comment out the add_matrix call having static_cast in parameter, i found one in ParameterCombination.cpp?
If so, shall i send single PR to address both(comment add_vector + add_matrix)?

While commenting the code, should i leave a remark with any issue no.?

karlnapf commented:
Don’t worry about Parameter combination, we will take care of that
